epochStart/metachain/systemSCs.go

```diff
@@ -158,23 +158,23 @@ func (s *systemSCProcessor) processWithNewFlags(
 func (s *systemSCProcessor) unStakeNodesWithNotEnoughFundsWithStakingV4(
```

Comment:
> this method seems to have a lot of code from unStakeNodesWithNotEnoughFundsWithStaking from legacySc

Yes, it is "the same" func, but refactored to be applied in staking v4. "Similarities" are expected; since it was decided to have more "duplicated" code with separate functions instead of lots of "if flagStakingV4.IsSet"

Reply:
>also this line in the legacySC
log.Debug("legacySystemSC: stakingV2", "enabled", epoch >= s.stakingV2EnableEpoch)
should take into consideration stakingV4 like here:
s.flagStakingV2Enabled.SetValue(epoch >= s.stakingV2EnableEpoch && epoch <= s.stakingV4InitEnableEpoch)

Right, very good catch; fixed it

Reply:
>for the current SetValue of flagStakingV2Enabled and flagStakingV4Enabled, both will be enabled in epoch with stakingV4 enabled. Is it ok?

Anything related to `flagStakingV2` will be available until `stakingV4InitEpoch`, including it.
`flagStakingV4Enabled` is enabled starting `stakingV4EnableEpoch`; but the whole chain is not completed until `stakingV4DistributeAuctionToWaiting` is also activated

Reply:
ok on all 3